components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -0,0 +1,208 @@
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addUsersToRole(roleID, newUserList, tenantDomain);
+        return roleDAO.removeUsersFromRole(roleID, deletedUserList, tenantDomain);
+    }
+
+    @Override
+    public List<GroupBasicInfo> getGroupListOfRole(String roleID, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getGroupListOfRole(roleID, tenantDomain);
+    }
+
+    @Override
+    public RoleBasicInfo updateGroupListOfRole(String roleID, List<String> newGroupNamesList,
+            List<String> deletedGroupNamesList, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addGroupsToRole(roleID, newGroupNamesList, tenantDomain);
+        return roleDAO.removeGroupsFromRole(roleID, newGroupNamesList, tenantDomain);
+    }
+
+    @Override
+    public List<String> getPermissionListOfRole(String roleID, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        String roleName = roleDAO.getRoleNameByID(roleID, tenantDomain);
+        try {
+            return getSelectedPermissions(getUserAdminProxy().getRolePermissions(roleName, tenantId));
+        } catch (UserAdminException e) {
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(),
+                    "An error occurred when retrieving permissions of role : " + roleID, e);
+        }
+    }
+
+    @Override
+    public RoleBasicInfo setPermissionsForRole(String roleID, List<String> permissions, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        String roleName = roleDAO.getRoleNameByID(roleID, tenantDomain);
+        try {
+            getUserAdminProxy().setRoleUIPermission(roleName, permissions.toArray(new String[0]));
+            return new RoleBasicInfo(roleID, roleName);
+        } catch (UserAdminException e) {
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(),
+                    "An error occurred when setting permissions for the role: " + roleID, e);
+        }
+    }
+
+    @Override
+    public boolean isExistingRole(String roleID, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.isExistingRoleID(roleID, tenantDomain);
+    }
+
+    /**
+     * Get the UserAdmin service.
+     *
+     * @return UserRealmProxy of UserAdmin service.
+     */
+    private UserRealmProxy getUserAdminProxy() {
+
+        UserRealm realm = (UserRealm) CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+        return new UserRealmProxy(realm);
```

Comment:
Shouldn't tenant domain being passed should be incorporated to pick the user realm

Reply:
Updated the PR with the discussed changes.